airbyte-ci/connectors/pipelines/pipelines/builds/python_connectors.py

```diff
@@ -15,13 +17,95 @@ class BuildConnectorImage(BuildConnectorImageBase):
     A spec command is run on the container to validate it was built successfully.
     """
 
+    DEFAULT_ENTRYPOINT = ["python", "/airbyte/integration_code/main.py"]
+    PATH_TO_INTEGRATION_CODE = "/airbyte/integration_code"
+
+    @property
+    def _build_connector(self):
+        if (
+            "connectorBuildOptions" in self.context.connector.metadata
+            and "baseImage" in self.context.connector.metadata["connectorBuildOptions"]
+        ):
+            return self._build_from_base_image
+        else:
+            return self._build_from_dockerfile
+
     async def _run(self) -> StepResult:
-        connector = await with_airbyte_python_connector(self.context, self.build_platform)
+        connector: Container = await self._build_connector()
         try:
             return await self.get_step_result(connector.with_exec(["spec"]))
         except QueryError as e:
             return StepResult(self, StepStatus.FAILURE, stderr=str(e))
 
+    def _get_base_container(self) -> Container:
+        base_image_name = self.context.connector.metadata["connectorBuildOptions"]["baseImage"]
+        self.logger.info(f"Building connector from base image {base_image_name}")
+        return self.dagger_client.container(platform=self.build_platform).from_(base_image_name)
+
+    async def _create_builder_container(self, base_container: Container) -> Container:
+        """Pre install the connector dependencies in a builder container.
+        If a python connectors depends on another local python connector, we need to mount its source in the container
+        This occurs for the source-file-secure connector for example, which depends on source-file
+
+        Args:
+            base_container (Container): The base container to use to build the connector.
+
+        Returns:
+            Container: The builder container, with installed dependencies.
+        """
+        ONLY_PYTHON_BUILD_FILES = ["setup.py", "requirements.txt", "pyproject.toml", "poetry.lock"]
```

Comment:
I think the list looks good generally for the cases we support, but is there any usecase for overriding these?

Also, do the dependencies on other local connectors still work with these whitelisted files? (assume so, but to confirm)

Reply:
I cant see a usecase for overriding these.

Basically this is defining "all ways to define python dependencies" and making sure they are mounted to the container.

If python adds yet another way, we would just extend this list instead of override.

also all the local dependency logic is still there and exists in `with_python_connector_installed`